### PR TITLE
Critical bug fix

### PR DESCRIPTION
**Pull request by RobinCK**
Please merge and update version of npm

### CODE REVIEW

**Comment by RobinCK** [comment]
fix for 'gulp-nightwatch'

File: browserstack.js [hunk 1]
```diff
@@ -35,30 +35,46 @@ var BrowserStackPatch = function () {
...
           frameworkPatch.addCapability('browserName', process.env.BSTACK_BROWSER);
         }
-        frameworkPatch.addCapability('os', process.env.BSTACK_OS);
-        frameworkPatch.addCapability('os_version', process.env.BSTACK_OS_VERSION);
-        frameworkPatch.addCapability('browser_version', process.env.BSTACK_BROWSER_VERSION);
-        frameworkPatch.addCapability('device', process.env.BSTACK_DEVICE);
-        frameworkPatch.trackFrameworkVersion();
 
-        frameworkPatch.seleniumHost('hub.browserstack.com', 80);
+        if(process.env.BSTACK_OS) {
```
Comment (RobinCK) [inline browserstack.js]: it's true

File: framework_patches/nightwatchPatch.js [hunk 2]
```diff
@@ -1,29 +1,31 @@
...
     },
+
     seleniumHost: function(host, port) {
       patchOptions['seleniumHost'] = host;
       patchOptions['seleniumPort'] = port;
     },
+
     trackFrameworkVersion: function() {
-      var nightwatchPackage = require(path.join(require.main.filename, '../../package.json'));
+      var nightwatchPackage = require(patchOptions.desiredCapabilities['browserstack.framework'] + '/package.json');
```
Comment (vedharish) [inline framework_patches/nightwatchPatch.js]: @RobinCK @pulkitsharma07 I think this implementation is better and we should use this

File: .editorconfig [hunk 3]
```diff
@@ -0,0 +1,10 @@
+root = true
```
Comment (vedharish) [inline .editorconfig]: @RobinCK Can you remove this file please.

We already use a linter and we would prefer to add a lint check on the master branch on github as a build.

Comment (RobinCK) [inline .editorconfig]: @vedharish editorconfig removed

File: browserstack.js (hunk 1 above)
Comment (pulkitsharma07) [inline browserstack.js]: Remove these too.

Comment (RobinCK) [inline browserstack.js]: @pulkitsharma07  I deleted only my

**Comment by pulkitsharma07** [comment]
Thanks.

**Comment by RobinCK** [comment]
You need to update npm version and publish

**Comment by maxcold** [comment]
@pulkitsharma07 can you release a new version please?

**Comment by daneshqureshi** [comment]
@RobinCK @maxcold, we have released the new version. Sorry for the delay.

Danesh Qureshi.
Team Browserstack.

**Comment by maxcold** [comment]
@daneshqureshi 👍

**Pull request merged**